portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java

```diff
@@ -37,9 +37,11 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         if (userService.isUserLoggedIn()) {
             Entity userInfoEntity = getUserInfoEntity();
 
-            String where = (String) userInfoEntity.getProperty("where");
-            // If there is nothing in the where filed, set it to contact.html by default.
-            if (where == null) {
+            String where = "/contact.html";
+            try {
+                where = (String) userInfoEntity.getProperty("where");
+            } catch (NullPointerException e) {
+                // If there is nothing in the where filed, set it to contact.html by default.
                 where = "/contact.html";
```

Comment:
this a Java thing? is the variable scope reduced to the try/catch block?

you're initializing a "where" variable in line 40 then redefining it in line 45

Reply:
Yes, if I declared the variable inside try I would only be able to access it inside the try. So I could access it later in the code, I declared the variable before the try-catch and then just updated its values inside that.

Reply:
give this a try @ https://www.tutorialspoint.com/compile_java_online.php

```
public class HelloWorld{

     public static void main(String []args){
        System.out.println("Hello World");
            String where = "/contact.html";
            try {
                System.out.println(where);
                throw new NullPointerException();
            } catch (NullPointerException e) {
                 System.out.println(where);
            }
     }
}
```

Reply:
I addressed this inside #57